docs: lifecycle-operator non-blocking deployment functionality (#3123)
keptn/lifecycle-toolkit: docs/docs/components/lifecycle-operator/keptn-non-blocking.md

---
comments: true
---

# Keptn non-blocking deployment functionality

Keptn provides an option to disable the default deployment blocking functionality
when pre-deployment tasks or evaluations (on KeptnApp or KeptnWorkload level) fail.

> By creating a [KeptnConfig](../../reference/crd-reference/config.md) resources and

Reviewer: ```suggestion
By populating a [KeptnConfig](../../reference/crd-reference/config.md) resource and
```

> setting the `.spec.blockDeployment` parameter of to `false` the blocking
> behavior for Keptn is disabled and therefore all applications will get deployed
> to the cluster whether the pre-deployment tasks or evaluations fail.

Reviewer: ```suggestion
setting the `.spec.blockDeployment` parameter to `false`, the blocking
behavior for Keptn is disabled and therefore all applications are deployed
to the cluster even if the pre-deployment tasks or evaluations fail.
```

This behavior is valuable if you want to execute a dry-run of the

> tasks/evaluations for the application, but still have your application deployed
> to the cluster not depending on the results of the pre-checks.

Reviewer: ```suggestion
tasks/evaluations for the application but have your application deployed
to the cluster regardless of the results of the pre-checks.
```

If the checks of the application fail, the state of the deployment phase

> are marked as `Warning` and you are able to inspect which

Reviewer: ```suggestion
is marked as `Warning` and you are can inspect which
```
